pkg/discovery/service.go

```diff
@@ -133,10 +133,14 @@ func (discovery *Discovery) refresh(ctx context.Context) error {
 			return ctx.Err()
 		}
 
+		if node.Id.IsZero() {
```

Comment:
Why would there be a zero node in the cache?

Reply:
This was a holdover from some previous testing, I'll remove it.